packages/react/src/functions/apply-patch-with-mutation.test.ts

```diff
@@ -0,0 +1,55 @@
+import { applyPatchWithMinimalMutationChain } from './apply-patch-with-mutation';
+
+describe('applyPatchWithMinimalMutationChain', () => {
```

Comment:
😄

Reply:
wrote these tests bc I didn't trust my implementation from 2 years ago. turns out it worked as expected, couldn't find any bugs, but hey we have more tests now :D